File: ruby/easy/67_add_binary.rb
# 67. Add Binary
# Easy
#
# Given two binary strings a and b, return their sum as a binary string.

def add_binary(a, b)
  (a.to_i(2) + b.to_i(2)).to_s(2)
end

puts add_binary('11', '1')       # 100
puts add_binary('1010', '1011')  # 10101

File: go/easy/67_add_binary.go
// 67. Add Binary
// Easy
//
// Given two binary strings a and b, return their sum as a binary string.
package main

import (
	"math/big"
)

func addBinary(a string, b string) string {
	aa, _ := new(big.Int).SetString(a, 2)
	bb, _ := new(big.Int).SetString(b, 2)

	res := &big.Int{}
	res.Add(aa, bb)

	return res.Text(2)
}
